source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/SettlementReport/SettlementReportResultServiceTests.cs

```diff
@@ -0,0 +1,78 @@
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports.Model;
+using Energinet.DataHub.Wholesale.CalculationResults.UnitTests.Infrastructure.SqlStatements;
+using Energinet.DataHub.Wholesale.Common.Models;
+using FluentAssertions;
+using Moq;
+using NodaTime;
+using Xunit;
+using Xunit.Categories;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.UnitTests.Infrastructure.SettlementReport;
+
+[UnitTest]
+public class SettlementReportResultServiceTests
+{
+    private readonly Instant _somePeriodStart = Instant.FromUtc(2021, 3, 1, 10, 15);
+    private readonly Instant _somePeriodEnd = Instant.FromUtc(2021, 3, 31, 10, 15);
+    private readonly string[] _someGridAreas = { "123", "456", };
+    private readonly Table _someTable = TableTestHelper.CreateTableForSettlementReport(3);
+    private readonly List<string> _columnNames = new() { ResultColumnNames.GridArea, ResultColumnNames.BatchProcessType, ResultColumnNames.Time, ResultColumnNames.TimeSeriesType, ResultColumnNames.Quantity, };
+
+    [Theory]
+    [AutoMoqData]
+    public async Task GetRowsAsync_ReturnsExpectedNumberOfRows(Mock<ISqlStatementClient> mockSqlStatementClient)
+    {
+        // Arrange
+        mockSqlStatementClient.Setup(s => s.ExecuteSqlStatementAsync(It.IsAny<string>())).ReturnsAsync(_someTable);
+        var sut = new SettlementReportResultService(mockSqlStatementClient.Object);
+
+        // Act
+        var actual = await sut.GetRowsAsync(_someGridAreas, ProcessType.BalanceFixing, _somePeriodStart, _somePeriodEnd, null);
+
+        // Assert
+        actual.Count().Should().Be(_someTable.RowCount);
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public async Task GetRowsAsync_ReturnsExpectedData(Mock<ISqlStatementClient> mockSqlStatementClient)
+    {
+        // Arrange
+        var row = new[] { "123", "BalanceFixing", "2022-05-16T01:00:00.000Z", "non_profiled_consumption", "1.234" };
+        var table = new Table(_columnNames,  new List<string[]> { row });
+        mockSqlStatementClient.Setup(s => s.ExecuteSqlStatementAsync(It.IsAny<string>())).ReturnsAsync(table);
+        var sut = new SettlementReportResultService(mockSqlStatementClient.Object);
+
+        // Act
+        var actual = await sut.GetRowsAsync(_someGridAreas, ProcessType.BalanceFixing, _somePeriodStart, _somePeriodEnd, null);
+
+        // Assert
+        var actualRow = actual.First();
+        actualRow.Quantity.Should().Be(1.234m);
+        actualRow.Time.Should().Be(Instant.FromUtc(2022, 5, 16, 1, 0, 0));
+        actualRow.Resolution.Should().Be("PT15M");
+        actualRow.GridArea.Should().Be("123");
+        actualRow.SettlementMethod.Should().Be(SettlementMethod.NonProfiled);
+        actualRow.ProcessType.Should().Be(ProcessType.BalanceFixing);
+        actualRow.MeteringPointType.Should().Be(MeteringPointType.Consumption);
```

Comment:
How about instantiating an `expected` and then replace with the following? Then you would also be sure that the test will catch props added in the future.
```suggestion
        actual.First().Should().Be(expected);
```